mmv1/third_party/terraform/services/alloydb/data_soruce_alloydb_cluster_test.go

```diff
@@ -0,0 +1,62 @@
+package alloydb_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccAlloydbDatabaseClusterDatasourceConfig(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+		"network_name":  acctest.BootstrapSharedServiceNetworkingConnection(t, "alloydb-instance-mandatory-1"),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckAlloydbClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAlloydbDatabaseClusterDatasourceConfig(context),
+			},
+		},
+	})
+}
+
+func testAccAlloydbDatabaseClusterDatasourceConfig(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_alloydb_instance" "default" {
```

Comment:
Was there a reason you're adding an instance to the config as well? Seems like maybe the test works well enough without it (just the cluster)